test/integration/controllers/cooperation.spec.js

```diff
@@ -0,0 +1,259 @@
+const { serverCleanup, serverInit, stopServer } = require('~/test/setup')
+const { expectError } = require('~/test/helpers')
+const { DOCUMENT_NOT_FOUND, UNAUTHORIZED } = require('~/consts/errors')
+const testUserAuthentication = require('~/utils/testUserAuth')
+const TokenService = require('~/services/token')
+
+const Offer = require('~/models/offer')
+const User = require('~/models/user')
+const Category = require('~/models/category')
+const Subject = require('~/models/subject')
+const Cooperation = require('~/models/cooperation')
+
+const endpointUrl = '/cooperations/'
+const nonExistingCooperationId = '19cf23e07281224fbbee3241'
+const nonExistingOfferId = '6487b9adf05b60536374c152'
+
+let tutorUserData = {
+  role: ['tutor'],
+  firstName: 'albus',
+  lastName: 'dumbledore',
+  email: 'lovemagic@gmail.com',
+  password: 'supermagicpass123',
+  appLanguage: 'en',
+  isEmailConfirmed: true,
+  lastLogin: new Date().toJSON()
+}
+
+let studentUserData = {
+  role: ['student'],
+  firstName: 'harry',
+  lastName: 'potter',
+  email: 'potter@gmail.com',
+  password: 'supermagicpass123',
+  appLanguage: 'en',
+  isEmailConfirmed: true,
+  lastLogin: new Date().toJSON()
+}
+
+const testCooperationData = {
+  price: 99,
+  proficiencyLevel: 'Beginner',
+  additionalInfo:
+    "I don't like both Dark Arts and Voldemort that's why i want to learn your subject and became your student"
+}
+
+const testOfferData = {
+  authorRole: 'tutor',
+  price: 99,
+  proficiencyLevel: 'Beginner',
+  title: 'First-class teacher. Director of the Hogwarts school of magic',
+  description: 'I will teach you how to protect yourself and your family from dark arts',
+  languages: 'English',
+  FAQ: [{ question: 'Do you enjoy being a director of the Hogwarts?', answer: 'Actually yes, i really like it.' }]
+}
+
+const updateData = {
+  price: 100,
+  status: 'active'
+}
+
+describe('Cooperation controller', () => {
+  let app, server, accessToken, testOffer, testCooperation, testStudentUser, testTutorUser
+
+  beforeAll(async () => {
+    ;({ app, server } = await serverInit())
+  })
+
+  beforeEach(async () => {
+    accessToken = await testUserAuthentication(app, studentUserData)
+
+    testStudentUser = TokenService.validateAccessToken(accessToken)
+
+    testTutorUser = await User.create(tutorUserData)
+
+    const category = await Category.create({
+      name: 'Dark Magic',
+      categoryIcon: {
```

Comment:
there is no "categoryIcon" field in category model anymore

Reply:
yeap, you are right, i have fixed